### ISSUE TITLE
Another regression with OSM? value too long for type character varying(69)

### ISSUE THREAD

**Issue by robe2** [pramsey/pgsql-ogr-fdw#59]
Can't figure this one out why this fails on my 9.6 with ogr_fdw master branch but worked in ogr_fdw 1.0.1 (at least on my 9.5)

```
CREATE SERVER osm_dc FOREIGN DATA WRAPPER ogr_fdw
OPTIONS (
datasource 'C:/LocatePress/data/dc.osm.pbf',
format 'OSM'
);

CREATE SCHEMA fdwt;
IMPORT FOREIGN SCHEMA ogr_all
FROM SERVER osm_dc INTO fdwt;

CREATE EXTENSION hstore;

SELECT
DISTINCT unnest(akeys(other_tags::hstore)) As key
FROM fdwt.points
ORDER BY key;


````

IN my 9.5 this returns 454 rows in 6 secs.  On my new ogr_fdw (on my 9.6) I get this error:

> ERROR:  value too long for type character varying(69)
********** Error **********

> ERROR:value too long for type character varying(69)
SQL state: 22001

This is the file I'm using - http://www.postgis.us/downloads/pgr_1e/dc.osm.zip

Which is a file I grabbed from http://download.geofabrik.de/north-america/us/district-of-columbia.html

**Comment by robe2** [comment]
Oh and to add this does randomondo but again works fine on my 9.5 ogr_fdw 1.0.1 install.  I think we had this issue before with dates or something as I recall which you had fixed.

```
SELECT
DISTINCT other_tags
FROM fdwt.points;
```
Gives:

>  ERROR:  value too long for type character varying(39)
********** Error **********

> ERROR: value too long for type character varying(39)
SQL state: 22001

**Comment by pramsey** [comment]
After bringing in bigint support for FIDs, per @rouault, I don't see anything like this (I saw errors about integer overflow though, not varchar)

**Comment by robe2** [comment]
well now I get a different error - might be related to the compiler warnings I just submitted which looks like that whole PRI64 mess I ran into with postgis_topology (that GNU and VC++ have different string format defs) and I mingw tends to use both

```
SELECT
DISTINCT other_tags
FROM fdwt.points;

SELECT
DISTINCT unnest(akeys(other_tags::hstore)) As key
FROM fdwt.points
ORDER BY key;

```
Both now give:

> ERROR: invalid input syntax for integer: "64d"
> SQL state: 22P02

**Comment by rouault** [comment]
This is likely due to https://github.com/pramsey/pgsql-ogr-fdw/commit/e079facac95f976b79951a3f96e103a271759c0a . But "%I64d" should be fine with mingw, since this is interpreted by the runtime lib and the runtime lib on Windows should support that.

**Comment by robe2** [comment]
It's the snprintf (at least one).  I made a change (copying from postgis_topology)
at https://github.com/robe2/pgsql-ogr-fdw/commit/8d0fb7b2b95fc71ffaebf3f8929b52f45194a9c3
 and doing that gets rid of the compile warnings noted in #66  as well as fixing 64d issue so limit 10 example works.  

I still get an error when doing:

```
SELECT
DISTINCT unnest(akeys(other_tags::hstore)) As key
FROM fdwt.points
ORDER BY key;

```

ERROR:  value "2159692184" is out of range for type integer
********** Error **********

ERROR: value "2159692184" is out of range for type integer
SQL state: 22003
.
So perhaps I don't have the format thing set right.

**Comment by robe2** [comment]
Okay getting warmer.  This last bug seems to be an issue with fid being set to integer instead of bigint.

If I change my foreign table structure to:

```
CREATE FOREIGN TABLE fdwt.points
   (fid bigint ,
    geom postgis.geometry ,
    osm_id character varying ,
    name character varying ,
    barrier character varying ,
    highway character varying ,
    ref character varying ,
    address character varying ,
    is_in character varying ,
    place character varying ,
    man_made character varying ,
    other_tags character varying )
   SERVER osm_dc
   OPTIONS (layer 'points');

```

Then the DISTINCT unnest ... works fine..  I reran 

```

IMPORT FOREIGN SCHEMA ogr_all FROM SERVER osm_dc INTO fdwt;

```

Just to verify this is not fixed (still comes in as integer instead of bigint.

**Comment by robe2** [comment]
Okay I see you have fid hard-coded to integer.  In case of OSM data they use I think the osm_id for fid, so that has to be a long it.  But my tree is out of date now since @rouault  had that in https://github.com/pramsey/pgsql-ogr-fdw/pull/63

**Comment by robe2** [comment]
Correction I thought the fid int issue would be fixed by @rouault  #63  commit, but it seems there is still a hardcoding at

[Line 2379](https://github.com/pramsey/pgsql-ogr-fdw/blob/master/ogr_fdw.c#L2379)
